source/solvers/navier_stokes_base.cc

```diff
@@ -2084,6 +2085,91 @@ NavierStokesBase<dim, VectorType, DofsType>::write_checkpoint()
     }
 }
 
+template <>
+inline bool
+NavierStokesBase<2, TrilinosWrappers::MPI::Vector, IndexSet>::
+  is_locally_owned_dof(const unsigned int global_id)
+{
+  return this->locally_owned_dofs.is_element(global_id);
+}
+template <>
+inline bool
+NavierStokesBase<3, TrilinosWrappers::MPI::Vector, IndexSet>::
+  is_locally_owned_dof(const unsigned int global_id)
+{
+  return this->locally_owned_dofs.is_element(global_id);
+}
+template <>
+inline bool
+NavierStokesBase<2, TrilinosWrappers::MPI::BlockVector, std::vector<IndexSet>>::
+  is_locally_owned_dof(const unsigned int global_id)
+{
+  return this->dof_handler.locally_owned_dofs().is_element(global_id);
+}
+template <>
+inline bool
+NavierStokesBase<3, TrilinosWrappers::MPI::BlockVector, std::vector<IndexSet>>::
+  is_locally_owned_dof(const unsigned int global_id)
+{
+  return this->dof_handler.locally_owned_dofs().is_element(global_id);
+}
+
+template <int dim, typename VectorType, typename DofsType>
+void
+NavierStokesBase<dim, VectorType, DofsType>::
+  rescale_pressure_dofs_in_newton_update()
+{
+  // We skip the function if the factor has a value of 1
+  if (abs(pressure_scaling_factor - 1) < 1e-8)
+    return;
+
+  // We check if the vectors are the same, which would mean the newton_update
+  // was already rescaled
+  VectorType temp;
+  temp.reinit(this->locally_owned_dofs, this->mpi_communicator);
+
+  temp += rescaled_newton_update;
+  temp -= newton_update;
+  if (temp.norm_sqr() < 1e-8)
+    {
+      newton_update = rescaled_newton_update;
+      return;
+    }
+  TimerOutput::Scope t(this->computing_timer, "rescale_pressure");
```

Comment:
I am not sure this is the best way to do this. I mean, the Newton update is calculated when we solve the linear system. So instead why not use a flag? Like when you calculate the newton update the flag "new_newton_update" member of the base is set to true or something. This equality cheecking is prohibitive I feel.

Reply:
It was useful when I wasn't sure of when to call the rescaling function, but I agree that it's not the best design. I removed this part and only call rescale_pressure_dofs_in_newton_update() when necessary now.